src/super_gradients/training/sg_trainer/sg_trainer.py

```diff
@@ -150,7 +150,7 @@ def __init__(self, experiment_name: str, device: str = None, multi_gpu: Union[Mu
 
         # SET THE EMPTY PROPERTIES
         self.net, self.architecture, self.arch_params, self.dataset_interface = None, None, None, None
-        self.train_loader, self.valid_loader = None, None
+        self.train_loader, self.valid_loaders_dict = None, None
```

Comment:
#### Dict vs List
The issue with a list is that we don't have name for the validation sets. This means that
- the metrics names would be defined based on the index: `1/Acc` , `2/Acc`, ... (as apposed to `cars/Acc`, `animals/Acc` for instance with a dict)
- the `metric_to_watch` would also be defined based on index. This can be very confusing imo, and is pretty error prone

#### `valid_loader` vs `valid_loaders_dict`
I started like that, but I felt like this was very misleading because `valid_loader` would actually be a dictionary of Dataloader, so the name would not represent the object.
For backward compatibility, I added this property 
``` python
    @property
    def valid_loader(self) -> DataLoader:
        # Meant for backward compatibility
        if len(self.valid_loaders_dict) == 1:
            logger.warning("`valid_loader` is deprecated in favor of `valid_loaders_dict`.")
            return next(iter(self.valid_loaders_dict.values()))
        else:
            raise RuntimeError("`valid_loader` not defined when working with multiple validation loaders. Please use `valid_loaders_dict` instead.")
```


What do you think?

Reply:
My opinion is this: 99% users are using one validation data loader. And we must not make their life more complicated that it already is.
From the other end, multiple validation loaders is an advanced feature, so you may expect some increased level of proficiency of a user and commitment. 
You can have two options:
A) Ask 1% of users to spends 10 minutes instead of 1 minute configuring multiple data loaders
B) Ask rest 99% of users to flex a little 

IMO options A is much more preferrable